Yosemite/Yosemite/Stores/ProductStore.swift

```diff
@@ -90,18 +90,24 @@ public class ProductStore: Store {
                                   let excludedProductIDs,
                                   let shouldDeleteStoredProductsOnFirstPage,
                                   let onCompletion):
-            synchronizeProducts(siteID: siteID,
-                                pageNumber: pageNumber,
-                                pageSize: pageSize,
-                                stockStatus: stockStatus,
-                                productStatus: productStatus,
-                                productType: productType,
-                                productCategory: productCategory,
-                                sortOrder: sortOrder,
-                                productIDs: productIDs,
-                                excludedProductIDs: excludedProductIDs,
-                                shouldDeleteStoredProductsOnFirstPage: shouldDeleteStoredProductsOnFirstPage,
-                                onCompletion: onCompletion)
+            Task { @MainActor in
```

Comment:
In one of my test runs I did encounter a concurrency warning on `ProductAttribute+ReadOnlyConvertible` but I have not been able to reproduce it again. From my initial check, it seemed to come from here so I've tried to play around with removing the call to `@MainActor` in this task, but I saw no difference

```
[...]
/woocommerce-ios/Yosemite/Yosemite/Model/Storage/ProductAttribute+ReadOnlyConvertible.swift:24 Performing I/O on the main thread can cause hangs.
This is known to cause hangs for your users.
```

Since this sync calls both the mapper via `loadAllProducts` and later `upsertStoredProductsInBackground` in background threads, perhaps we are able to remove the MainActor tag if not needed 🤔

Reply:
Created an issue about the performance warning: WOOMOB-619

Reply:
> Did the stack trace start from accessing ResultsController.fetchedObjects? I've been seeing ProductAttribute+ReadOnlyConvertible warnings in trunk and the PR branch for a while.

I'm not sure, but now that you mention it I might have seen the same warning outside this PR as well before 🤔 

Thanks for logging it in!

> The @MainActor only puts the non-awaiting code on the main thread, in this case the completion callbacks. I think this is expected and how it works before, this PR just ensures the networking code, particularly the response decoding part, is not on the main thread.

👍